flytekit/types/structured/basic_dfs.py

```diff
@@ -24,6 +21,13 @@
     StructuredDatasetEncoder,
 )
 
+if typing.TYPE_CHECKING:
+    import pandas as pd
+    import pyarrow as pa
```

Comment:
RE: the coverage decrease here, we should probably configure Codecov to ignore blocks like these by adding a `.coveragerc` file containing the following:

```
[report]
exclude_lines = 
    pragma: no cover
    if TYPE_CHECKING:
```

See: https://github.com/nedbat/coveragepy/issues/831#issuecomment-517778185

Apparently it's not configurable through `codecov.yml` 🙈 : https://github.com/codecov/codecov-python/issues/107#issuecomment-793976712